### PR TITLE
feature: update the username and password of git-checkout action for safety

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
feature: update the username and password of git-checkout action for safety

#### Which issue(s) this PR fixes:
[cicd 流水线 git-checkout action 注入 token 代替 git_inner_user](https://erda.cloud/erda/dop/projects/387/issues/all?id=208149&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=TASK)

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      update the username and password of git-checkout action for safety        |
| 🇨🇳 中文    |     git-checkout action 注入 token 代替 git_inner_user         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkg/pipelinecms/generate_name.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021 Terminus, Inc.
```
Comment (sfwn) [inline pkg/pipelinecms/generate_name.go:1]: move pkg to modules/pipeline/pkg/cms.

dop can invoke pipeline's pkg.

File: modules/pipeline/services/pipelinesvc/create_v2.go [hunk 2]
```diff
@@ -49,11 +50,17 @@ func (s *PipelineSvc) CreateV2(req *apistructs.PipelineCreateRequestV2) (*spec.P
...
 		_p, err := s.RunPipeline(&apistructs.PipelineRunRequest{
-			PipelineID:        p.ID,
-			ForceRun:          req.ForceRun,
-			IdentityInfo:      req.IdentityInfo,
-			PipelineRunParams: req.RunParams,
+			PipelineID:             p.ID,
+			ForceRun:               req.ForceRun,
+			IdentityInfo:           req.IdentityInfo,
+			PipelineRunParams:      req.RunParams,
+			ConfigManageNamespaces: []string{pipelinecms.MakeUserOrgPipelineCmsNs(req.IdentityInfo.UserID, orgID)},
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/create_v2.go:63]: pipeline will never care about this logic.
dop need also request with these configManage namespaces at create time.
If a same key exist in multi configMananges, latter will overwrite the former. So run-time's namespace will overwrite create-time's.

File: modules/pipeline/services/pipelinesvc/run.go [hunk 3]
```diff
@@ -61,6 +61,13 @@ func (s *PipelineSvc) RunPipeline(req *apistructs.PipelineRunRequest) (*spec.Pip
 		return nil, err
 	}
 
+	if len(req.ConfigManageNamespaces) > 0 {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/run.go:64]: append to string slice directly is ok.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/pipelinecms/generate_name.go (hunk 1 above)
Comment (littlejiancc) [reply pkg/pipelinecms/generate_name.go]: done

File: modules/pipeline/services/pipelinesvc/create_v2.go (hunk 2 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/create_v2.go]: done

File: modules/pipeline/services/pipelinesvc/run.go (hunk 3 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/run.go]: done

File: modules/pipeline/services/pipelinesvc/create_v2.go [hunk 4]
```diff
@@ -50,10 +50,11 @@ func (s *PipelineSvc) CreateV2(req *apistructs.PipelineCreateRequestV2) (*spec.P
...
 		_p, err := s.RunPipeline(&apistructs.PipelineRunRequest{
-			PipelineID:        p.ID,
-			ForceRun:          req.ForceRun,
-			IdentityInfo:      req.IdentityInfo,
-			PipelineRunParams: req.RunParams,
+			PipelineID:             p.ID,
+			ForceRun:               req.ForceRun,
+			IdentityInfo:           req.IdentityInfo,
+			PipelineRunParams:      req.RunParams,
+			ConfigManageNamespaces: req.ConfigManageNamespaces,
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/create_v2.go:57]: No need this line. It's duplicated with create.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/create_v2.go (hunk 4 above)
Comment (littlejiancc) [reply modules/pipeline/services/pipelinesvc/create_v2.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**